### PR TITLE
solana/program_v2: Clean up payments for passes

### PR DESCRIPTION
**Pull request by JulianSpring**
Fixed some error handling renames and added more tests

### CODE REVIEW

File: solana/program_v2/programs/gateway_v2/src/util.rs [hunk 1]
```diff
@@ -32,25 +35,25 @@ pub trait OnChainSizeWithArg<Arg> {
...
+pub fn get_network_fees(fees: &[NetworkFees], mint: Pubkey) -> Result<&NetworkFees, NetworkErrors> {
+    fees.iter().find(|&&x| x.token == mint).ok_or(NetworkErrors::FeesNotProvided)
 }
 
 /// calculate_network_and_gatekeeper_fee
 /// Returns two fees in the correct unit
 /// First result returns the fee for the network_fee
 /// Second result returns the gatekeeper fee
+/// Split -> percent
 pub fn calculate_network_and_gatekeeper_fee(fee: u64, split: u16) -> (u64, u64) {
```
Comment (JulianSpring) [inline solana/program_v2/programs/gateway_v2/src/util.rs:51]: Forgot to rename split to percent will do that now

File: solana/program_v2/packages/tests/src/pass-suite/issue.test.ts [hunk 2]
```diff
@@ -14,16 +15,19 @@ import {
   makeAssociatedTokenAccountsForIssue,
   setUpAdminNetworkGatekeeper,
 } from '../test-set-up';
+import { setGatekeeperFlagsAndFees } from '../util/lib';
 
-describe('issue', () => {
+describe.only('issue', () => {
```
[the remark this answers is not in the document]
Comment (JulianSpring) [reply solana/program_v2/packages/tests/src/pass-suite/issue.test.ts:20]: Yeah should be removed, need to set up linting for this 🙏

File: solana/program_v2/programs/gateway_v2/src/instructions/gatekeeper/expire_pass.rs [hunk 3]
```diff
@@ -21,17 +20,9 @@ pub fn expire_pass(ctx: Context<PassExpire>) -> Result<()> {
...
-        Some(fee) => fee.expire,
-        None => return Err(error!(GatekeeperErrors::GatekeeperFeeNotProvided)),
-    };
-
-    let raw_network_fee = match get_network_fees(&network.fees, *mint_address) {
-        Some(fee) => fee.expire,
-        None => return Err(error!(NetworkErrors::NetworkFeeNotProvided)),
-    };
-    let fees = calculate_network_and_gatekeeper_fee(raw_gatekeeper_fee, raw_network_fee);
+    let absolut_fee = get_gatekeeper_fees(&gatekeeper.token_fees, *mint_address)?.expire;
```
Comment (DiosAkhilleus) [inline solana/program_v2/programs/gateway_v2/src/instructions/gatekeeper/expire_pass.rs:23]: check spelling here

File: solana/program_v2/programs/gateway_v2/src/instructions/gatekeeper/issue_pass.rs [hunk 4]
```diff
@@ -1,53 +1,45 @@
...
-        Some(fee) => fee.issue,
-        None => return Err(error!(GatekeeperErrors::GatekeeperFeeNotProvided)),
-    };
-
-    let raw_network_fee = match get_network_fees(&network.fees, *mint_address) {
-        Some(fee) => fee.issue,
-        None => return Err(error!(NetworkErrors::NetworkFeeNotProvided)),
-    };
-    let fees = calculate_network_and_gatekeeper_fee(raw_gatekeeper_fee, raw_network_fee);
+    let absolut_fee = get_gatekeeper_fees(&gatekeeper.token_fees, *mint_address)?.issue;
```
Comment (DiosAkhilleus) [inline solana/program_v2/programs/gateway_v2/src/instructions/gatekeeper/issue_pass.rs:24]: spelling again, is this meant to be "absolut"?

File: solana/program_v2/programs/gateway_v2/src/instructions/gatekeeper/refresh_pass.rs [hunk 5]
```diff
@@ -20,18 +20,9 @@ pub fn refresh_pass(ctx: Context<PassRefresh>) -> Result<()> {
...
-        Some(fee) => fee.refresh,
-        None => return Err(error!(GatekeeperErrors::GatekeeperFeeNotProvided)),
-    };
-
-    let raw_network_fee = match get_network_fees(&network.fees, *mint_address) {
-        Some(fee) => fee.refresh,
-        None => return Err(error!(NetworkErrors::NetworkFeeNotProvided)),
-    };
-    let fees = calculate_network_and_gatekeeper_fee(raw_gatekeeper_fee, raw_network_fee);
+    let absolut_fee = get_gatekeeper_fees(&gatekeeper.token_fees, *mint_address)?.refresh;
```
Comment (DiosAkhilleus) [inline solana/program_v2/programs/gateway_v2/src/instructions/gatekeeper/refresh_pass.rs:23]: Same as above

File: solana/program_v2/programs/gateway_v2/src/instructions/gatekeeper/verify_pass.rs [hunk 6]
```diff
@@ -21,17 +20,9 @@ pub fn verify_pass(ctx: Context<PassVerify>) -> Result<()> {
...
-        Some(fee) => fee.verify,
-        None => return Err(error!(GatekeeperErrors::GatekeeperFeeNotProvided)),
-    };
-
-    let raw_network_fee = match get_network_fees(&network.fees, *mint_address) {
-        Some(fee) => fee.verify,
-        None => return Err(error!(NetworkErrors::NetworkFeeNotProvided)),
-    };
-    let fees = calculate_network_and_gatekeeper_fee(raw_gatekeeper_fee, raw_network_fee);
+    let absolut_fee = get_gatekeeper_fees(&gatekeeper.token_fees, *mint_address)?.verify;
```
Comment (DiosAkhilleus) [inline solana/program_v2/programs/gateway_v2/src/instructions/gatekeeper/verify_pass.rs:23]: same here

File: solana/program_v2/packages/tests/src/pass-suite/issue.test.ts [hunk 7]
```diff
@@ -176,14 +232,16 @@ describe('issue', () => {
 
     gatekeeperService
       .issue(
-        account,
+        passAccount,
```
[the remark this answers is not in the document]
Comment (JulianSpring) [reply solana/program_v2/packages/tests/src/pass-suite/issue.test.ts:235]: Ah yeah not needed thanks!

**Review by william-brooks — APPROVED** [review]

**Pull request merged**